src/linear_algebra/finite_dimensional.lean

```diff
@@ -850,6 +850,61 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma findim_eq_zero_iff_subsingleton [finite_dimensional K V] :
+  findim K V = 0 ↔ subsingleton V :=
+begin
+  rw [← not_iff_not, ← not_nontrivial_iff_subsingleton, not_not],
+  change findim K V ≠ 0 ↔ _,
+  rw ← pos_iff_ne_zero,
+  exact findim_pos_iff
+end
+
+lemma eq_zero_of_findim_eq_zero [finite_dimensional K V]
+  (hV : findim K V = 0) (x : V) : x = 0 :=
+begin
+  rw findim_eq_zero_iff_subsingleton at hV,
+  cases hV,
+  exact hV x 0,
+end
+
+lemma eq_zero_of_dim_eq_zero (hV : dim K V = 0) (x : V) : x = 0 :=
+begin
+  haveI := finite_dimensional_of_dim_eq_zero hV,
+  rw ← findim_eq_dim at hV,
+  norm_cast at hV,
+  exact eq_zero_of_findim_eq_zero hV x
+end
```

Comment:
This lemma already exists, its `dim_zero_iff_forall_zero`

Reply:
Can you add a `dim_zero_iff_subsingleton` lemma right next to the one that exists?

Reply:
Will do once #6376 is merged